evap/evaluation/models.py

```diff
@@ -447,6 +448,47 @@ def is_archiveable(self):
     def was_evaluated(self, request):
         self.course_evaluated.send(sender=self.__class__, request=request, semester=self.semester)
 
+    @property
+    def final_grade_documents(self):
+        from evap.grades.models import GradeDocument
+        return self.grade_documents.filter(type=GradeDocument.FINAL_GRADES)
+
+    @property
+    def midterm_grade_documents(self):
+        from evap.grades.models import GradeDocument
+        return self.grade_documents.exclude(type=GradeDocument.FINAL_GRADES)
+
+    @classmethod
+    def update_courses(cls):
+        from evap.evaluation.tools import send_publish_notifications, send_evaluation_started_notifications
+        today = datetime.date.today()
+
+        courses_new_in_evaluation = []
+        grade_document_courses = []
+        evaluation_results_courses = []
+
+        for course in cls.objects.all():
+            try:
+                if course.state == "approved" and course.vote_start_date <= today:
+                    course.evaluation_begin()
+                    course.save()
+                    courses_new_in_evaluation.append(course)
+                elif course.state == "inEvaluation" and course.vote_end_date < today:
+                    course.evaluation_end()
+                    if course.is_fully_reviewed():
+                        course.review_finished()
+                        if not course.is_graded or course.final_grade_documents.exists():
+                            course.publish()
+                            evaluation_results_courses.append(course)
+                    if course.final_grade_documents.exists():
+                        grade_document_courses.append(course)
```

Comment:
this might be wrong now. if it has grades, an email for that should've been sent already.